lychee-lib/src/types/base.rs

```diff
@@ -27,18 +29,28 @@ impl Base {
                 let full_path = path.join(link);
                 Url::from_file_path(full_path).ok()
             }
+            Self::RootPath(_path) => {
+                // this is unused currently because joining on RootPath is handled by create_uri_from_file_path
+                unreachable!()
+            }
```

Comment:
I wonder if we should make `RootPath` a separate type. It feels unrelated to `Base`; the `unreachable` code path could be seen as an indicator of that. 
Also see my other comment for an explanation.